src/store/initial-state.js

```diff
@@ -17,7 +17,6 @@ export const createInitialState = () => ({
   prettyName: settings.prettyName.default,
   ignoreLargeWarning: false,
   loading: {
-    graph: false,
```

Comment:
This state cannot be deleted as this is also utilised for the display of the spinner during graph calculations ( refer to   `selectors/loading.js` and the tests outlined under `graph.test.js`

Reply:
Having this in breaks one of the tests in `src/components/app/app.test.js`:

The failure: 

```
App › updates the store › but does not override non-pipeline values
```

It expects `"graph": false` but receives `"graph": true`, and I think that's because we not longer prevent the graph from loading based on the status of the font loading. This is why I took it out.

Should I also remove the other relevant `getGraphLoading` lines from `selectors/loading.js` or leave the graph property in there and get the tests to pass? I just don't know if it's relevant anymore, that's the thing.

Reply:
Yea it's because the layout calculation is triggered in this test (`fontLoaded` was what prevented it before, which feels like a hack at best and maybe a coincidence at worst). I wonder if there is anyway to wait for the calculate to be done before assert? Otherwise we can probably modify the expectation to assert for `"graph": true`

Reply:
Yes, exactly.

> I wonder if there is anyway to wait for the calculate to be done before assert?

We could maybe try a timeout or something. That doesn't seem great though. Changing to `"graph": true` in the `loading` object solves it. I'd rather do that.

Reply:
I wonder if a search for webfont on kedro-viz slack channel would yield anything interesting? I recall seeing it being discussed on slack.

Reply:
Good call. Here's the relevant discussion: 

<img width="1369" alt="image" src="https://user-images.githubusercontent.com/16869061/153048677-087e6cd2-a030-43b5-9f8b-e44110ee4207.png">

The statement from Richard makes me think we don't need this graph check anymore. He says:

> I don’t think it currently does delay booting the whole app, it only delays running a layout calculation and displaying the whole graph. The font is loaded asynchronously while the rest of the app is loaded, but then it doesn’t do layout/graph rendering until after the font is loaded because it’d just have to update it again anyway.

I think we can safely do the layout/graph rendering without messing up the layout because of the font loading. Meaning: we don't have to block it due to fonts.

Plus, now we're using `<link rel="preconnect" ... />` which was essentially a wish of his.

Reply:
I agree there's no point in blocking the loading of the graph due to fonts ( in the old setup, the graph can still load with the fallback font when, say, there's no internet and is unable to retrieve web fonts, so there's no need to delay it). That said though, `loading.graph` is not just used for checking the loading of the fonts - it is still used to check if the `calculateGraph` action is still executing, and hence, display the spinner. 

In regards to the test, I might approach it to mocking the dispatch of the action to update the graph loader rather than changing the value in `prepareNonPipelineState` as it's a once off situation for this test ( to simulate the state when the graph finishes loading) - changing it to `graph: true` within the `prepareNonPipelineState` might have further implications to the entire app. ( I know it's a much more painful way, but it's much better to mock for a one off situation rather than changing the state initialisation logic.)

Reply:
I think I follow you but I'm not 100% sure. I've made the changes [here](https://github.com/kedro-org/kedro-viz/pull/731/commits/cadc1b3ff7d9f0971f6253179ecd0f583db7b553) by adding a TOGGLE_GRAPH_LOADING test and then removing the non-pipeline values one, because the `graph` value in that doesn't match anymore due to the removal of the `fontLoading`. Let me know if this is what you meant.

Reply:
just checked the code, and haha nope, this isn't what I meant at all ( esp with the deletion of the test), but regardless it is good to add the test for the reducer. 

what I meant initially was in this test is as follows:

```
test('but does not override non-pipeline values', () => {
      const wrapper = shallow(<App data={demo} />);
      wrapper.setProps({ data: spaceflights });
      // mock the dispatch of the action for graph loading back to false
      expect(getState(wrapper)).toMatchObject(prepareNonPipelineState({}));
    });
  });
```

alternatively, a quick getaround to this ( other than deleting the test) could be just manually assign the nonpipelinestate object to false:

```
test('but does not override non-pipeline values', () => {
      // mock the initial pipeline state except for the graph loading state given the state updates immediately on mount
      const mockNonPipelineState = Object.assign(prepareNonPipelineState({}), {loading:{graph: false}})

      const wrapper = shallow(<App data={demo} />);
      wrapper.setProps({ data: spaceflights });
      expect(getState(wrapper)).toMatchObject(mockNonPipelineState);
    });
  });
```

Reply:
Ohhhh ok, I get it. Please check! [Commit is here](https://github.com/kedro-org/kedro-viz/pull/731/commits/627c0f57024848307d4d80fe65a7c08a7ae9964c).